src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -275,6 +275,7 @@ typedef struct {
     uint64_t max_atomic_count;
     uint64_t max_compare_atomic_count;
     uint64_t max_fetch_atomic_count;
+    bool mpi_acc_valid;
```

Comment:
It seems my old comments have magically disappeared. My questions were:

1. can't we combine this information with `atomic_valid` instead of introducing a new flag?
2. Why MPI-level validity check is necessary here, which is supposed to be checked at MPI layer?

Reply:
I would like to separate the MPI semantics validity and OFI capability flags (e.g., `atomic_valid `-> for atomic write, `fetch_atomic_valid` -> for atomic fetch (+ write), `compare_atomic_valid`). A valid MPI accumulate might not be a valid OFI atomic op, similarly a valid OFI atomic op might not be a valid MPI accumulate. Separating these flags can help understand when/why a HW atomic can not be used.

We actually do not check validity here. The `mpi_acc_valid ` flag is used to exclude the case where an <op, dtype> is both an invalid MPI accumulate and an invalid OFI atomics.  I wrote explanation in commit comment.

Reply:
For the first part: I thought OFI atomics was a subset of MPI atomics. Maybe I'm wrong. I now approve your approach here.

For the second part, I'll need to understand more about how this value is used later in the code.